handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1555,10 +1558,34 @@ public final synchronized SSLParameters getSSLParameters() {
 
     @Override
     public final synchronized void setSSLParameters(SSLParameters sslParameters) {
-        super.setSSLParameters(sslParameters);
-
         int version = PlatformDependent.javaVersion();
         if (version >= 7) {
+            if (sslParameters.getAlgorithmConstraints() != null) {
+                throw new IllegalArgumentException("AlgorithmConstraints are not supported.");
+            }
+
+            if (version >= 8) {
+                Collection<SNIMatcher> matchers = sslParameters.getSNIMatchers();
+                if (matchers != null && !matchers.isEmpty()) {
+                    throw new IllegalArgumentException("SNIMatchers are not supported.");
```

Comment:
@Scottmitch FYI we can easily support SNIMatchers tho with a few changes in netty-tcnative. But I would like to fix this after this is pulled in as this needs more testing etc.